src/realm/query_expression.hpp

```diff
@@ -251,6 +252,51 @@ struct Pow {
     typedef T type;
 };
 
+template <class T>
+struct Size {
+    int64_t operator()(T) const
+    {
+        return 1;
+    }
+    typedef T type;
+};
+
+template <>
+struct Size<StringData> {
+    int64_t operator()(StringData v) const
+    {
+        return v.size();
+    }
+    typedef StringData type;
+};
+
+template <>
+struct Size<BinaryData> {
+    int64_t operator()(BinaryData v) const
+    {
+        return v.size();
+    }
+    typedef BinaryData type;
+};
+
+template <>
+struct Size<ConstTableRef> {
+    int64_t operator()(ConstTableRef v) const
+    {
+        return v->size();
+    }
+    typedef ConstTableRef type;
+};
+
+template <>
+struct Size<ConstLinkViewRef> {
+    int64_t operator()(ConstLinkViewRef v) const
+    {
+        return v->size();
+    }
+    typedef ConstLinkViewRef type;
+};
+
```

Comment:
No. StringData and BinaryData are passed by value objects whereas the other types are pointer type references.